src/lib/storage/table.cpp

```diff
@@ -79,6 +90,35 @@ void Table::_add_value_segment_to_chunk(std::shared_ptr<Chunk>& chunk, const std
   });
 }
 
-void Table::compress_chunk(ChunkID chunk_id) { throw std::runtime_error("Implement Table::compress_chunk"); }
+void Table::_add_dictionary_segment_to_chunk(std::shared_ptr<Chunk>& chunk, const std::string& type,
+                                             const std::shared_ptr<BaseSegment>& value_segment) {
+  resolve_data_type(type, [&](const auto data_type_t) {
+    using ColumnDataType = typename decltype(data_type_t)::type;
+    const auto dictionary_segment = std::make_shared<DictionarySegment<ColumnDataType>>(value_segment);
+    chunk->add_segment(dictionary_segment);
+  });
+}
+
+void Table::compress_chunk(ChunkID chunk_id) {
+  auto compressed_chunk = std::make_shared<Chunk>();
+  const auto& chunk = get_chunk(chunk_id);
+  const auto chunk_column_count = chunk.column_count();
+
+  std::vector<std::thread> threads;
+
+  for (auto column = 0; column < chunk_column_count; column++) {
+    const auto segment_type = _column_types[column];
+    auto segment = chunk.get_segment(static_cast<ColumnID>(column));
+
+    threads.emplace_back(
+        std::thread(&Table::_add_dictionary_segment_to_chunk, this, std::ref(compressed_chunk), segment_type, segment));
```

Comment:
If you use emplace, do not construct std:thread, instead just pass the parameters of the constructor of std:thread.